src/test/atlas_connectivity.rs

```diff
@@ -13,13 +13,9 @@ async fn run_test(uri_env_var: &str, resolver_config: Option<ResolverConfig>) {
     };
 
     let uri_string = uri.to_string_lossy();
-    let options = match resolver_config {
-        Some(resolver_config) => {
-            ClientOptions::parse_with_resolver_config(uri_string.as_ref(), resolver_config).await
-        }
-        None => ClientOptions::parse(uri_string.as_ref()).await,
-    }
-    .expect("uri parsing should succeed");
+    let options = ClientOptions::parse_uri(uri_string.as_ref(), resolver_config)
```

Comment:
In this case, using `parse_uri` is actually way more readable than calling the public methods; either that method didn't exist when I originally wrote these tests, or I forgot that it did.

Reply:
I'd personally prefer if the tests used the public API as much as possible, even if they are just thin wrappers. If we're not running the async tests as part of the sync tests, why not just conditionally compile the entire test module? That way we can continue to use the public `parse` without per-test conditional compilation.

Reply:
Updated to disable the module when sync is enabled

Reply:
Turned out that didn't quite work due to test module functionality being used outside of the module; I've pushed again with more selective disabling